bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/RuntimeUtils.java

```diff
@@ -247,4 +248,11 @@ private static boolean isInvalidBallerinaValue(Object value) {
 
     private RuntimeUtils() {
     }
+
+    public static Type getEffectiveType(Type type) {
+        if (type.getTag() == TypeTags.INTERSECTION_TAG) {
+            return ((BIntersectionType) type).getEffectiveType();
+        }
+        return type;
```

Comment:
Codecov warnings?

Reply:
Fixed with current changes

Reply:
With the current changes, the issue has been resolved.